photutils/detection/tests/test_peakfinder.py

```diff
@@ -68,12 +65,10 @@ def test_npeaks(self):
 
     def test_border_width(self):
         """Test border exclusion."""
-        with catch_warnings(NoDetectionsWarning) as warning_lines:
+        with pytest.warns(NoDetectionsWarning,
+                          match='No local peaks were found'):
             tbl = find_peaks(PEAKDATA, 0.1, box_size=3, border_width=3)
             assert tbl is None
-            assert len(warning_lines) > 0
```

Comment:
No need for this anymore as `pytest.warns` will throw error if the warning is not there.